src/commands/availability/availability.js

```diff
@@ -0,0 +1,368 @@
+        let timeFrom = interaction.options.get('time-from')?.value;
+        let timeTo = interaction.options.get('time-to')?.value;
+        const reason = interaction.options.get('reason')?.value;
+
+        if (dateTo && !dateFrom)
+            throw new ScheduleError('Please select a start date.');
+        if (timeTo && !timeFrom)
+            throw new ScheduleError('Please select a start time.');
+
+        // Create a start and end dayjs obj (Default times to 0:00 if empty)
+        const startUnavail = dayjs(`2024 ${dateFrom} ${timeFrom ? timeFrom : '0:00'}`);
+        const endUnavail = dayjs(`2024 ${dateTo} ${timeTo ? timeTo : '23:59'}`);
+
+        if (!dayjs(startUnavail).isValid && !dayjs(endUnavail).isValid)
+            throw new ScheduleError('Enter dates and times in proper formats');
+
+        const unavail = createUnavailability(startUnavail, endUnavail, reason);
+
+        // Print data for now
+        let reply = [
+            '```',
+            JSON.stringify(unavail, undefined, 2),
+            '```',
+        ].join('\n');
+
+        await interaction.editReply({ content: reply });
+
+        // Save data to file
+        saveUnavailability(userId, userTag, unavail, path);
+    } catch (error) {
+        if (error.name === 'ScheduleError') {
+            await interaction.editReply({
+                ephemeral: true,
+                content:   `*${error.message}*`
+            });
+        } else {
+            console.log(error);
+            await interaction.editReply({
+                ephemeral: true,
+                content:   `*Issue running command*`
+            });
+        }
+    }
+};
+
+// set-availability
+const setAvail = async (interaction) => {
+    const userId = interaction?.user?.id;
+    const userTag = interaction?.user?.tag;
+
+    await interaction.deferReply({ ephemeral: true });
+
+    // command should include a user
+    if (userId === undefined || userTag === undefined) {
+        await interaction.editReply({
+            ephemeral: true,
+            content:   'Could not process command'
+        });
+        return;
+    }
+
+    try {
+        const timeFrom = interaction.options.get('time-from')?.value;
+        const timeTo = interaction.options.get('time-to')?.value;
+        const rawDays = interaction.options.get('days')?.value;
+
+        if (!timeFrom || !timeTo)
+            throw new ScheduleError('Enter both start AND end times');
+
+        // if (!rawDays)
+        //     throw new ScheduleError('Enter available days');
+
+        // Parse the day list into an array
+        const days = parseDaysList(rawDays ? rawDays : 'daily');
+
+        // Create a start and end dayjs obj (arbitrary day used, does not affect time result)
+        const startAvail = dayjs(`2024 5-20 ${timeFrom}`);
+        const endAvail = dayjs(`2024 8-9 ${timeTo}`);
+
+        if (!dayjs(startAvail).isValid && !dayjs(endAvail).isValid)
```

Comment:
This error checking should be in a separate method like createAvailability. The only purpose of a command is to call a method